File: from-3101-to-3200/3163-string-compression-iii/solution.go
func compressedString(word string) string {
    result := []string{}
    curC := word[0]
    cnt := 1
    for i := 1; i < len(word); i ++ {
        if word[i] == curC && cnt < 9 {
            cnt += 1
        } else {
            result = append(result, fmt.Sprintf("%d%c", cnt, curC))
            curC = word[i]
            cnt = 1
        }
    }
    result = append(result, fmt.Sprintf("%d%c", cnt, curC))
    return strings.Join(result, "")
}

File: from-3101-to-3200/3163-string-compression-iii/solution.py
class Solution:
    def compressedString(self, word: str) -> str:
        result = ""
        curC = word[0]
        cnt = 1
        for c in word[1:]:
            if c == curC and cnt < 9:
                cnt += 1
            else:
                result += f"{cnt}{curC}"
                curC = c
                cnt = 1
        result += f"{cnt}{curC}"
        return result